If headers_install is executed from a deep/long directory structure, the
shell's maximum argument length can be execeeded. Create one .input-files
file per sub module to allow multi threaded make calls.

--- a/scripts/Makefile.headersinst
+++ b/scripts/Makefile.headersinst
@@ -72,7 +72,8 @@ printdir = $(patsubst $(INSTALL_HDR_PATH
 quiet_cmd_install = INSTALL $(printdir) ($(words $(all-files))\
                             file$(if $(word 2, $(all-files)),s))
       cmd_install = \
-        $(CONFIG_SHELL) $< $(installdir) $(input-files); \
+        xargs $(CONFIG_SHELL) $< $(installdir) < $(installdir)/.input-files; \
+        rm $(installdir)/.input-files; \
         for F in $(wrapper-files); do                                   \
                 echo "\#include <asm-generic/$$F>" > $(installdir)/$$F;    \
         done;                                                           \
@@ -101,6 +102,7 @@ targets += $(install-file)
 $(install-file): scripts/headers_install.sh $(input-files) FORCE
 	$(if $(unwanted),$(call cmd,remove),)
 	$(if $(wildcard $(dir $@)),,$(shell mkdir -p $(dir $@)))
+	@echo $(input-files) > $(dir $@)/.input-files
 	$(call if_changed,install)
 
 else
